### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by IamYunusAli**
In this pull request me @IamYunusAli  and my partner @Alusp have done the following things
- Create Tests for the following methods. `factorial`, `reverse` and  `fizzbuzz`
- Create a class called `Solver`.
- Create a method called `factorial` that takes one argument, an integer `N`, and returns the for that number. The factorial is the multiplication of all integers from 1 to `N` and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if `word` is `"hello"` it returns `"olleh"`).
- Create a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string. The returned string is constructed following these rules:
  - When `N` is divisible by 3, return `"fizz"`.
  - When `N` is divisible by 5, return `"buzz"`.
  - When `N` is divisible by 3 and 5, return `"fizzbuzz"`.
  - Any other case, return `N` as a string (e.g. say `N` is `7` then return `"7"`).

### CODE REVIEW

File: spec/solve_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative './spec_helper'
+
+describe Solver do
+  it 'test for the solver factorial method' do
+    solve = Solver.new
+    expect(solve.factorial(5)).to be_kind_of(Integer)
+    expect(solve.factorial(5)).to eq(120)
+    expect(solve.factorial(0)).to eq(1)
+    expect(solve.factorial(1)).to eq(1)
```
Comment (omar-muhamad) [inline spec/solve_spec.rb:9]: - [ ] Please add a test case for the factorial of a `Negative number` which raises an exception as an expectation to cover all test cases.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @IamYunusAli and @Alusp,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @TEAM

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request

- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
